rcldotnet/rcldotnet.c

```diff
@@ -46,6 +48,27 @@ void native_rcl_reset_error() { rcl_reset_error(); }
 bool native_rcl_ok() { return rcl_ok(); }
 
 int32_t native_rcl_create_node_handle(void **node_handle, const char *name, const char *namespace) {
+
+  // Validate the node name
+  int validation_result;
+  size_t invalid_index;
+  rmw_ret_t rmw_ret =
+    rmw_validate_node_name(name, &validation_result, &invalid_index);
+
+  if (rmw_ret != RMW_RET_OK) {
+    if (rmw_ret == RMW_RET_BAD_ALLOC) {
+      fprintf(stderr, "Memory error: %s\n", rmw_get_error_string());
```

Comment:
The error message won't be shown if the application is not run from the command line.

Reply:
Ok. I prefer to keep this message in stderr, if you don't mind. As I added an exception also, I think it is redundant but not annoying.

Reply:
I don't think it adds much, for example, if you're building a GUI, you won't see the error, so I'd prefer to remove the `fprintf` calls.

Reply:
Ok. Fixed